core/dbt/cli/main.py

```diff
@@ -178,10 +181,11 @@ def docs(ctx, **kwargs):
 @requires.preflight
 @requires.profile
 @requires.project
+@requires.runtime_config
+@requires.manifest
 def docs_generate(ctx, **kwargs):
     """Generate the documentation website for your project"""
-    config = RuntimeConfig.from_parts(ctx.obj["project"], ctx.obj["profile"], ctx.obj["flags"])
-    task = GenerateTask(ctx.obj["flags"], config)
+    task = GenerateTask(ctx.obj["flags"], ctx.obj["runtime_config"])
```

Comment:
sorry that I'm catching this post-merge but - should `GenerateTask` be passed the manifest?

Reply:
I think so, it is inheriting `CompileTask`, the reason it need manifest is that generate task would go to the warehouse and fetch all info about the models in current dbt project.

Reply:
Good catch! `docs generate` also performs a full `compile` of all manifest nodes — unless `--select` is passed, in which case only those nodes; or if `--no-compile` is passed, in which case it doesn't

Reply:
Looks like the issue is being addressed by @aranke in the [cli test work here](https://github.com/dbt-labs/dbt-core/pull/6723/files#diff-43832707bfec8cdaa49c13f004d9b3296585827f7e68ef07fba7ad1d5e6eb7c6R193).